internal/webhook/external/github.go

```diff
@@ -0,0 +1,152 @@
+package external
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	gh "github.com/google/go-github/v71/github"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+var (
+	// errMissingSignature is returned when the 'X-Hub-Signature-256'
+	// header is not found or empty.
+	errMissingSignature = errors.New("missing signature")
+	// errSecretUnset is returned when the 'GH_WEBHOOK_SECRET'
+	// environment variable is empty.
+	errSecretUnset = errors.New("secret is unset")
+)
```

Comment:
These are each used in exactly one place. It would be more straightforward to just return `errors.New("missing signature")` `errors.New("secret is unset")` from those places.